org.scala-ide.sdt.core.tests/src/org/scalaide/core/testsetup/SDTTestUtils.scala

```diff
@@ -347,12 +352,26 @@ object SDTTestUtils extends HasLogger {
 
     entries ++= withSourceFolders(project, javaProject)
 
-    entries += JavaCore.newContainerEntry(Path.fromPortableString(SdtConstants.ScalaLibContId))
     javaProject.setRawClasspath(entries.toArray[IClasspathEntry], null)
+    javaProject
+  }
+
+  private[core] def internalCreateProjectInWorkspace(projectName: String, withSourceFolders: SrcPathOutputEntry): ScalaProject = {
+    def withScalaFolders(project: IProject, jProject: IJavaProject) =
+      withSourceFolders(project, jProject) ++ Seq(JavaCore.newContainerEntry(Path.fromPortableString(SdtConstants.ScalaLibContId)))
+    def addScalaNature(project: IProject) = {
+      val description = project.getDescription
+      description.setNatureIds(description.getNatureIds ++ Array(SdtConstants.NatureId))
+      project.setDescription(description, null)
+      project
+    }
 
-    ScalaPlugin().getScalaProject(project)
+    ScalaPlugin().getScalaProject(addScalaNature(internalCreateJavaProjectInWorkspace(projectName, withScalaFolders).getProject))
```

Comment:
createJavaProjectInWorkspace is a public one and SDTTestUtils is a trait so is safer to create private method even if it is the whole body of public method